data/src/main/java/com/linkedin/data/collections/CheckedMap.java

```diff
@@ -541,8 +556,60 @@ private void notifyChangeListenersOnClear(Set<? extends K> keys)
     void onUnderlyingMapChanged(K key, V value);
   }
 
+  private void purgeStaleChangeListeners()
+  {
+    if (_changeListenerReferenceQueue != null && _changeListenerReferenceQueue.poll() != null)
+    {
+      // Clear finalized weak references.
+      while (_changeListenerReferenceQueue.poll() != null)
+      {
+        // Do nothing, as we are just clearing the reference queue.
+      }
+      // Now iterate over change listeners and purge stale references.
+      WeakListNode<ChangeListener<K, V>> node = _changeListenerHead,  prev = null;
+      while (node != null)
+      {
+        if (node._object.get() == null)
+        {
+          if (prev == null)
+          {
+            _changeListenerHead = node._next;
```

Comment:
So when the head points to the next node, the original head will be GCed by jvm?

Reply:
yes.